packages/types/src/models/V1beta1ProviderSpecSecret.ts

```diff
@@ -0,0 +1,117 @@
+/**
+ * Forklift API
+ * Migration toolkit for virtualization (Forklift) API definitions.
+ *
+ * The version of the OpenAPI document: 2.4.0
+ * Contact: kubev2v-dev@redhat.com
+ *
+ * NOTE: This class is auto generated by OpenAPI Generator (https://openapi-generator.tech).
+ * https://openapi-generator.tech
+ * Do not edit the class manually.
+ */
+
+import { exists } from "../runtime";
+/**
+ * References a secret containing credentials and other confidential information.
+ * @export
+ * @interface V1beta1ProviderSpecSecret
+ */
+export interface V1beta1ProviderSpecSecret {
```

Comment:
Is this interface supposed to align with the legacy type `ISecret`? https://github.com/kubev2v/forklift-console-plugin/blob/604a8878cc5f6fedf354b747797ccfec9eeca13a/packages/legacy/src/queries/types/secrets.types.ts#L3-L14

Is there anywhere where the data shape of the provider's secret data is supposed to be defined in a CRD?  Maybe I'm missing something?

Reply:
no, this is the `name`+`namespace` struct used byt the provider to point to the secret

the actual secret struct is a dictionary of `string: string`  and is not documenteds in a CRD,
so it's not part of the API, and the backend can use any `key` of type `string`
and any `value` of tyep string

when we read the secret we should document somewhere that we assume some keys to exist, and what there value should mean.

https://konveyor.github.io/forklift/migratingvms/migratecli/
actually there are several different key: value combinations for each provider, the rules are not in a CRD:
https://github.com/kubev2v/forklift/blob/960eefe94e1c60fda9171c654039c3ec37480e40/pkg/controller/provider/validation.go#L207

Reply:
@yaacov please make it a separate commit:)